### PR TITLE
Update release instructions to include rc branch

### PR DESCRIPTION
**Pull request by rnpridgeon**

### CODE REVIEW

File: tools/RELEASE.md [hunk 1]
```diff
@@ -114,13 +117,13 @@ An alternative is to push branch and tags separately:
     $ git push --dry-run --tags origin v0.11.4rc1
 
 
-## 5. Wait for CI builds
+## 6. Wait for CI builds
 
 Monitor travis-ci builds by looking atthe *tag* build at
 [travis-ci]https://travis-ci.org/confluentinc/confluent-kafka-python
```
Comment (edenhill) [inline tools/RELEASE.md]: markdown link format is:
`[linktext](url)`

File: tools/RELEASE.md [hunk 2]
```diff
@@ -154,7 +156,12 @@ and librdkafka versions:
     py: ('0.11.4', 721920) c: ('0.11.4-RC1', 722121)
 
 
-## 8. Upload packages to PyPi
+## 9. Merge RC branch into master
+
+Once all builds are successful and artifacts have been validated proceed with merging the changes into Master.
```
Comment (edenhill) [inline tools/RELEASE.md]: "and the PR has been approved"

File: tools/RELEASE.md [hunk 3]
```diff
@@ -24,10 +24,13 @@ The following guide uses `v0.11.4rc1` as version for a release candidate,
 replace as necessary with your version or remove `rc..` suffix for the
 final release.
 
+## 1. Create a rc branch
```
[the remark this answers is not in the document]
Comment (edenhill) [reply tools/RELEASE.md]: Add that a PR should be created that needs to be reviewed by a team member.

File: tools/RELEASE.md [hunk 4]
```diff
@@ -154,7 +156,12 @@ and librdkafka versions:
     py: ('0.11.4', 721920) c: ('0.11.4-RC1', 722121)
 
 
-## 8. Upload packages to PyPi
+## 9. Merge RC branch into master
+
+Once all builds are successful and artifacts have been validated proceed with merging the changes into Master. 
+Be sure to squash any unrelated commits (typos, build errors) to keep the master commit log clean.
```
Comment (edenhill) [inline tools/RELEASE.md]: You typically squash/fixup **related** commits.

So maybe something like:
`Be sure to tidy up the commit log in the PR branch by squashing related commits (typos, build fixes, etc)`.

File: tools/RELEASE.md [hunk 5]
```diff
@@ -24,10 +24,13 @@ The following guide uses `v0.11.4rc1` as version for a release candidate,
 replace as necessary with your version or remove `rc..` suffix for the
 final release.
 
+## 1. Create a rc branch
 
-## 1. Update librdkafka and OpenSSL versions
+    git checkout -b v0.11.6rc
```
[the remark this answers is not in the document]
Comment (rnpridgeon) [reply tools/RELEASE.md]: All or nothing. Do you want 0.11.4 or 0.11.6

File: tools/RELEASE.md (hunk 1 above)
Comment (rnpridgeon) [reply tools/RELEASE.md]: Not sure I understand. Are you requesting an additional improvement on the docs?

File: tools/RELEASE.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (edenhill) [reply tools/RELEASE.md]: v0.11.4 seems to be a smaller change, go with that.

File: tools/RELEASE.md (hunk 1 above)
Comment (edenhill) [reply tools/RELEASE.md]: the link to travis-ci is incorrect, check the rendered version of RELEASE.md and you'll see the link doesnt work. So use the above format.

File: tools/RELEASE.md (hunk 4 above)
Comment (rnpridgeon) [reply tools/RELEASE.md]: Thanks I was having trouble figuring out how to correctly word that.

File: tools/RELEASE.md [hunk 6]
```diff
@@ -153,8 +155,18 @@ and librdkafka versions:
     $ python -c 'import confluent_kafka as ck ; print "py:", ck.version(), "c:", ck.libversion()'
     py: ('0.11.4', 721920) c: ('0.11.4-RC1', 722121)
 
+## 10. Open a release PR
+
+Once all builds are successful and artifacts have been validated proceed with opening a PR.
```
Comment (edenhill) [inline tools/RELEASE.md]: You can do the re-build cycle dance in the PR branch, don't need to wait with creating a PR.

File: tools/RELEASE.md [hunk 7]
```diff
@@ -153,8 +155,18 @@ and librdkafka versions:
...
+This PR will need to approved by a team member prior to proceeding with the release.
+
+## 11. Merge RC branch into master
+
+Once all builds are successful, artifacts have been validated and a team member has approved the PR
+proceed with merging the changes into Master. Be sure to tidy up the commit log in the PR branch by
```
Comment (edenhill) [inline tools/RELEASE.md]: "master", not "Master". Branch names are case sensitive

**Pull request merged**